src/PointwiseFunctions/AnalyticData/BnsInitialData/SpecInitialData.hpp

```diff
@@ -0,0 +1,335 @@
+      gr::Tags::Lapse<DataType>,
+      ::Tags::deriv<gr::Tags::Lapse<DataType>,
+                    tmpl::integral_constant<size_t, 3>, Frame::Inertial>,
+      gr::Tags::Shift<DataType, 3>,
+      ::Tags::deriv<gr::Tags::Shift<DataType, 3>,
+                    tmpl::integral_constant<size_t, 3>, Frame::Inertial>,
+      Tags::RotationalShift<DataType>,
+      Tags::DerivLogLapseTimesDensityOverSpecificEnthalpy<DataType>,
+      Tags::RotationalShiftStress<DataType>>;
+  template <typename DataType>
+  using tags = tmpl::append<background_tags<DataType>, variable_tags<DataType>>;
+
+  struct DataDirectory {
+    using type = std::string;
+    static constexpr Options::String help = {
+        "Path to a directory of data produced by SpEC. The directory is "
+        "expected to contain 'GrDomain.input' and 'Vars*.h5' files for all the "
+        "subdomains in GrDomain.input."};
+  };
+
+  struct DensityCutoff {
+    using type = double;
+    static constexpr Options::String help =
+        "Where the density is below this cutoff the fluid variables are set to "
+        "vacuum (zero density, pressure, energy and velocity, unit Lorentz "
+        "factor and enthalpy). "
+        "During the evolution, atmosphere treatment will typically kick in and "
+        "fix the value of the fluid variables in these regions. Therefore, "
+        "it makes sense to set this density cutoff to the same value as the "
+        "atmosphere density cutoff.";
+  };
+
+  using options = tmpl::list<
+      DataDirectory,
+      hydro::OptionTags::InitialDataEquationOfState<true, ThermodynamicDim>,
+      DensityCutoff,
+      BnsInitialData::InitialData::TovStar::OrbitalAngularVelocity,
+      BnsInitialData::Tags::OptionTags::EulerEnthalpyConstant>;
+
+  static constexpr Options::String help = {"Initial data generated by SpEC"};
+
+  SpecInitialData() = default;
+  SpecInitialData(const SpecInitialData& rhs);
+  SpecInitialData& operator=(const SpecInitialData& rhs);
+  SpecInitialData(SpecInitialData&& /*rhs*/) = default;
+  SpecInitialData& operator=(SpecInitialData&& /*rhs*/) = default;
+  ~SpecInitialData() override = default;
+
+  SpecInitialData(std::string data_directory,
+                  std::unique_ptr<equation_of_state_type> equation_of_state,
+                  double density_cutoff, double orbital_angular_velocity,
+                  double euler_enthalpy_constant);
+
+  auto get_clone() const
+      -> std::unique_ptr<elliptic::analytic_data::Background>;
+
+  /// \cond
+  explicit SpecInitialData(CkMigrateMessage* msg);
+  using PUP::able::register_constructor;
+  WRAPPED_PUPable_decl_template(SpecInitialData);
+  /// \endcond
+  const equation_of_state_type& equation_of_state() const {
+    return *equation_of_state_;
+  }
+  // The velocity potential is used in the initial guess
+  template <typename DataType>
+  tuples::TaggedTuple<Tags::VelocityPotential<DataType>> variables(
```

Comment:
Move the definition of these `variables` functions to the cpp file. You can also remove the templating on `DataType` and just use `DataVector` (if you want).